tests-legacy/Unit/Core/SpecificPrice/SpecificPriceFormatterTest.php

```diff
@@ -0,0 +1,244 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace LegacyTests\Unit\Core\SpecificPrice;
+
+use Context;
+use Currency;
+use Language;
+use LegacyTests\Unit\ContextMocker;
+use LocalizationPack;
+use PrestaShop\PrestaShop\Adapter\Product\PriceFormatter;
+use PrestaShopBundle\Cache\LocalizationWarmer;
+use Tests\TestCase\SymfonyIntegrationTestCase;
+use PrestaShop\PrestaShop\Core\SpecificPrice\SpecificPriceFormatter;
+
+class SpecificPriceFormatterTest extends SymfonyIntegrationTestCase
+{
+    /**
+     * @var ContextMocker
+     */
+    protected $contextMocker;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        self::installTestedLanguagePacks();
+    }
+
+    protected function tearDown()
+    {
+        global $kernel;
+        unset($kernel);
+        parent::tearDown();
+        $this->contextMocker->resetContext();
+    }
+
+    protected static function installTestedLanguagePacks()
+    {
+        $countries = [
+            'us',
+            'fr',
+        ];
+        $cacheDir = _PS_CACHE_DIR_ . 'sandbox' . DIRECTORY_SEPARATOR;
+
+        foreach ($countries as $country) {
+            $xmlContent = (new LocalizationWarmer(_PS_VERSION_, $country))
+                ->warmUp($cacheDir);
+
+            (new LocalizationPack())->loadLocalisationPack($xmlContent, false, true);
+        }
+    }
+
+    /**
+     * @dataProvider specificPricesProvider
+     *
+     * @param $price
+     * @param $taxRate
+     * @param $ecotaxAmount
+     * @param $currencyData
+     * @param $specificPrices
+     * @param $isTaxIncluded
+     * @param $expected
+     */
+    public function testFormatSpecificPrice(
+        $price,
+        $taxRate,
+        $ecotaxAmount,
+        $currencyData,
+        $specificPrices,
+        $isTaxIncluded,
+        $expected
+    ) {
+        $context = Context::getContext();
+
+        $currency = new Currency();
+        $currency->active = true;
+        $currency->conversion_rate = $currencyData['conversion_rate'];
+        $currency->sign = $currencyData['sign'];
+        $currency->iso_code = $currencyData['code'];
+        $context->currency = $currency;
+
+        $language = new Language();
+        $language->iso_code = 'EN';
+        $language->locale = 'en-US';
+        $context->language = $language;
+
+        $specificPriceFormatter = new SpecificPriceFormatter($specificPrices[0], $isTaxIncluded, $context);
+        $formattedSpecificPrice = $specificPriceFormatter->formatSpecificPrice($price, $taxRate, $ecotaxAmount);
+
+        $priceFormatter = new PriceFormatter();
```

Comment:
I'm not sure I understand this test You actually compare both behaviours and test if they match?
But then it's not really a unit test

Reply:
There is only one method tested, it's a legacy method that returns an array, I'm checking that the **discount** and **save** values are matching with expectations.

Behaviors are specified in the provider (if we can call that a behavior). This is a legacy test that I had to move because I moved the method that is tested.